src/PSSparseServerTask.cpp

```diff
@@ -107,7 +107,7 @@ bool PSSparseServerTask::process_send_lr_gradient(const Request& req, std::vecto
 
   model_lock.lock();
   if (task_config.get_use_adagrad()) {
-    lr_model->sgd_update_adagrad(
+    lr_model->sgd_update_momentum(
```

Comment:
This doesn't seem to make sense. We should update our configuration code to allow saying we want to use momentum (or nesterov).

I suggest removing the _use_adagrad_configuration value and adding one like _opt_method_ that can have values: _sgd_, _nesterov_, _adagrad_, _momentum_. Check Configuration.h/cpp